scipy/special/_c99compat.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Portable versions of some C99 functions. In cmath they are only
+ * available in C++11 and the npy_math versions don't work well (they
+ * get undef'd by cmath; see gh-5689). Implementations based on
+ * npy_math.h and friends.
+ */
+#ifndef C99COMPAT_H
+#define C99COMPAT_H
+
+#include <Python.h>
+#include <numpy/numpyconfig.h>
+
+
+int
+sc_signbit(double x)
```

Comment:
So my understanding (maybe wrong) was: the CI machine has C99, so it hits

`#define npy_signbit(x) signbit((x))`

in `npy_math.h`. Then we hit `#undef signbit` in `cmath`. But then there's no C++11, so we end up having no `signbit`. Maybe this could also be fixed by shuffling the includes, though that sounds kind of brittle.